interfaces/udev/spec.go

```diff
@@ -93,6 +108,14 @@ func (c byTagAndSnippet) Less(i, j int) bool {
 
 // Snippets returns a copy of all the snippets added so far.
 func (spec *Specification) Snippets() (result []string) {
+	// If the interface controls it's own device cgroup, then
+	// we don't want to enforce a device cgroup, which is only turned on if
+	// there are udev rules, and as such we don't want to generate any udev
+	// rules
+
+	if spec.ControlsDeviceCgroup() {
```

Comment:
this merits  a backend_test.go  test  with a snap  having plugs of a test interface that would TagDevice and one that declares to control the device cgroup.

Reply:
I added a test here similar to the other tests in backed_test.go that just checks the udev rules file doesn't exist when `stat`'d.